### PR TITLE
feat: Programmatically open "more filters" dropdown in Horizontal Filter Bar

### PR DESCRIPTION
**Pull request by geido**
### SUMMARY
This PR adds the ability to open the "more filters" dropdown in the horizontal filter bar while refactoring the code to better differentiate between focus and hover actions on native filters.

### AFTER

https://user-images.githubusercontent.com/60598000/204836413-190dc840-a2b3-4c1f-8385-86e31837947f.mp4

### TESTING INSTRUCTIONS
1. Enable HORIZONTAL_FILTER_BAR feature flag
2. Open a dashboard
3. Set the filter bar to horizontal
4. Click on a filter that is currently within the "more filters" dropdown
5. The dropdown should open and focus the filter

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [x] Required feature flags: HORIZONTAL_FILTER_BAR
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/dashboard/components/gridComponents/Tabs.jsx [hunk 1]
```diff
@@ -340,9 +340,10 @@ export class Tabs extends React.PureComponent {
     const { tabIndex: selectedTabIndex, activeKey } = this.state;
 
     let tabsToHighlight;
-    if (nativeFilters?.focusedFilterId) {
-      tabsToHighlight =
-        nativeFilters.filters[nativeFilters.focusedFilterId].tabsInScope;
+    const highlighterFilterId =
+      nativeFilters?.focusedFilterId || nativeFilters?.hoveredFilterId;
+    if (highlighterFilterId) {
+      tabsToHighlight = nativeFilters.filters[highlighterFilterId]?.tabsInScope;
```
Comment (michael-s-molina) [inline superset-frontend/src/dashboard/components/gridComponents/Tabs.jsx:346]:
```suggestion
    const highlightedFilterId =
      nativeFilters?.focusedFilterId || nativeFilters?.hoveredFilterId;
    if (highlightedFilterId) {
      tabsToHighlight = nativeFilters.filters[highlightedFilterId]?.tabsInScope;
```

File: superset-frontend/src/dashboard/components/nativeFilters/FilterBar/FilterControls/utils.ts [hunk 2]
```diff
@@ -21,8 +21,21 @@ import { Dispatch } from 'react';
...
 
+export const dispatchHoverAction = debounce(
+  (dispatch: Dispatch<any>, id?: string) => {
+    if (id) {
+      dispatch(setHoveredNativeFilter(id));
+    } else {
+      dispatch(unsetHoveredNativeFilter());
+    }
+  },
+  300,
```
Comment (michael-s-molina) [inline superset-frontend/src/dashboard/components/nativeFilters/FilterBar/FilterControls/utils.ts:36]: There's a `FAST_DEBOUNCE` in `src/constants.ts`

File: superset-frontend/src/dashboard/util/useFilterFocusHighlightStyles.ts [hunk 3]
```diff
@@ -49,8 +49,9 @@ const useFilterFocusHighlightStyles = (chartId: number) => {
     dashboardFilters,
   );
 
-  const focusedNativeFilterId = nativeFilters.focusedFilterId;
-  if (!(focusedFilterScope || focusedNativeFilterId)) {
+  const highlighterFilterId =
+    nativeFilters?.focusedFilterId || nativeFilters?.hoveredFilterId;
+  if (!(focusedFilterScope || highlighterFilterId)) {
```
Comment (michael-s-molina) [inline superset-frontend/src/dashboard/util/useFilterFocusHighlightStyles.ts:54]:
```suggestion
  const highlightedFilterId =
    nativeFilters?.focusedFilterId || nativeFilters?.hoveredFilterId;
  if (!(focusedFilterScope || highlightedFilterId)) {
```

File: superset-frontend/src/dashboard/util/useFilterFocusHighlightStyles.ts [hunk 4]
```diff
@@ -67,9 +68,9 @@ const useFilterFocusHighlightStyles = (chartId: number) => {
     pointerEvents: 'auto',
   };
 
-  if (focusedNativeFilterId) {
+  if (highlighterFilterId) {
```
Comment (michael-s-molina) [inline superset-frontend/src/dashboard/util/useFilterFocusHighlightStyles.ts:71]:
```suggestion
  if (highlightedFilterId) {
```

File: superset-frontend/src/dashboard/util/useFilterFocusHighlightStyles.ts [hunk 5]
```diff
@@ -67,9 +68,9 @@ const useFilterFocusHighlightStyles = (chartId: number) => {
...
     if (
-      nativeFilters.filters[focusedNativeFilterId]?.chartsInScope?.includes(
+      nativeFilters.filters[highlighterFilterId]?.chartsInScope?.includes(
```
Comment (michael-s-molina) [inline superset-frontend/src/dashboard/util/useFilterFocusHighlightStyles.ts:73]:
```suggestion
      nativeFilters.filters[highlightedFilterId]?.chartsInScope?.includes(
```

**Review by michael-s-molina — APPROVED** [review]
LGTM

**Comment by michael-s-molina** [comment]
@geido Can you rebase this?

**Review by kgabryje — CHANGES_REQUESTED** [review]
I noticed 1 regression. In filter cards we have "Dependent on" section, with links to filters that given filter is dependent on. Previously clicking on that link would highlight the parent filter and currently nothing happens
image

**Review by kgabryje — APPROVED** [review]
Lgtm!

**Pull request merged**